### DISCUSSION TITLE
Missing imports

### DISCUSSION THREAD

**Discussion by Bub283** [ghostselfbot/ghost#9] [Help] [answered]
I installed every import needed thru pip install -r requirements.txt but im still missing a few imports:

Traceback (most recent call last):
  File "C:\Users\Bub\Documents\ghost-3.3.0\ghost.py", line 22, in <module>
    from utils import console
  File "C:\Users\Bub\Documents\ghost-3.3.0\utils\__init__.py", line 12, in <module>
    from .privnote import Privnote
  File "C:\Users\Bub\Documents\ghost-3.3.0\utils\privnote.py", line 6, in <module>
    import Crypto
ModuleNotFoundError: No module named 'Crypto'

**Comment by Bub283** [comment]
C:\Users\Bub\Documents\ghost-3.3.0>pip list
Package            Version
------------------ -----------
aiohappyeyeballs   2.4.3
aiohttp            3.10.10
aiosignal          1.3.1
art                6.3
async-timeout      4.0.3
attrs              24.2.0
certifi            2024.8.30
cfscrape           2.0.0
charset-normalizer 3.4.0
colorama           0.4.6
discord-protos     0.0.2
discord.py-self    2.0.0
Faker              30.8.2
frozenlist         1.5.0
idna               3.10
multidict          6.1.0
pillow             11.0.0
pip                21.2.3
propcache          0.2.0
protobuf           5.28.3
psutil             6.1.0
pypresence         4.3.0
pystyle            2.9
python-dateutil    2.9.0.post0
requests           2.32.3
setuptools         57.4.0
six                1.16.0
typing_extensions  4.12.2
urllib3            2.2.3
yarl               1.17.1
list of all the libs i have installed

**Comment by bennyscripts** [comment]
Here are some troubleshooting tips, can you also let me know what version of Python you are running.
- What version of python are you running?
- The first two import errors are from the utils. Try creating a virtual environment and running it in there. (https://mothergeo-py.readthedocs.io/en/latest/development/how-to/venv-win.html)
- The Crypto module can be installed using `pip install pycrypto`.

**Comment by Bub283** [comment]
Pycrypto is outdated and doesnt install.
![image](https://github.com/user-attachments/assets/55d597b2-5103-4cab-83a6-9030f06cde7b)
The new lib is apparently "pycryptodome".

**Reply by bennyscripts** [reply]
Oh yea sorry that's right, I've got pycryptodome installed so give that a go. Also try running Ghost in a virtual environment so none of your other python modules can interfere.

**Comment by Bub283** [comment, marked as answer]
Running in a virtual environment fixed the issue, thanks
